File: MODULE_3/EcomApplication/src/main/java/org/northernarc/ecomapplication/model/Product.java
package org.northernarc.ecomapplication.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Data
@Table(name="Product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String brand;
    private String category;
    private double cost;

    @OneToMany(mappedBy = "product",cascade = CascadeType.PERSIST)
    @JsonIgnore
    private List<Order_Item> orderItems;
}

File: MODULE_6/day5/EcomFrontEndDev/src/app/dto/product.ts
export interface ProductRequestDTO {
	name: string;
	brand: string;
	category: string;
	cost: number;
	description?: string;
}

export interface ProductResponseDTO {
	id: number;
	name: string;
	brand: string;
	category: string;
	cost: number;
	description?: string;
}

export interface ProductUpdateDTO {
	name?: string;
	brand?: string;
	category?: string;
	cost?: number;
	description?: string;
}
